sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -130,7 +140,14 @@ public static ClientPipeline Create(
         int perCallIndex = index;
 
         // Add retry policy.
-        policies[index++] = options.RetryPolicy ?? ClientRetryPolicy.Default;
+        if (loggingOptions == null || loggingOptions.ShouldUseDefaultRetryPolicy())
+        {
+            policies[index++] = options.RetryPolicy ?? new ClientRetryPolicy();
```

Comment:
Why would we not use the default retry policy if we have determined we should use the default retry policy?  (I am assuming this is the intent of the `ShouldUseDefaultRetryPolicy`, but pls correct me if I am wrong about this)

Reply:
That was a bug, fixed now